src/subgraph/actions/listDeposits.ts

```diff
@@ -0,0 +1,35 @@
+import { ApolloClient } from "@apollo/client/core";
+import { Deposit } from "../../types";
+import * as queries from "../queries";
+
+export const listDeposits = async <T>(
+  apolloClient: ApolloClient<T>,
+  poolAddress: string
+): Promise<Deposit[]> => {
+  const collection: Deposit[] = [];
+
+  const queryResult = await apolloClient.query<Deposit[]>({
```

Comment:
The subgraph is not returning a `Deposit` type.

It's returning something similar, but also, very different.

Take a look at the `zns-sdk` on how it handles data transformation from Subgraph => SDK world.

You're going to need to create a bunch of types/interfaces to represent what Subgraph returns and then a transformation method to convert into what the SDK wants.